src/Paket.Core/ProjectFile.fs

```diff
@@ -1337,6 +1338,18 @@ type ProjectFile with
 
     member this.FindReferencesFile() = this.FindCorrespondingFile Constants.ReferencesFile
 
+    member this.FindLocalizedLanguageNames() =
+        let pattern = @"Properties\\Resources.(?<language>\w+(-\w+)?).resx"
+        let includeattribute = "Include"
+        this.ProjectNode
+        |> getDescendants "EmbeddedResource"
+        |> List.filter (hasAttribute includeattribute)
+        |> List.map (fun n -> n.Attributes.[includeattribute])
+        |> List.map (fun a -> a.Value)
+        |> List.map (fun a -> Regex.Match(a, pattern, RegexOptions.ExplicitCapture))
```

Comment:
it feels like a helper local function to handle the regex logic (return `Some "the identified language"` or `None` with single function call) would avoid mixing list operation with multistep processing of regex (which is more imperative thing).